src/options.c

```diff
@@ -433,8 +437,12 @@ options_parse_display(char *optarg)
    if (length > 256) {
      length = 256;
    }
-   new_display = gib_emalloc(length);
-   strncpy(new_display, optarg, length);
+   new_display = malloc(length);
+   if (! new_display) {
+     fprintf(stderr, "Unable to allocate display: %s", strerror(errno));
+     exit(EXIT_FAILURE);
+   }
+   strcpy(new_display, optarg);
```

Comment:
Please revert this to the function `strncpy`
The var. `length` has a maximum of 256 and `optarg` could be greater overlapping the memory hosted by `malloc`

Reply:
I reverted it